### ISSUE TITLE
After Node restart, command rejected (NOT_FOUND) error

### ISSUE THREAD

**Issue by bhowmikdebarshi** [zeebe-io/zeebe#3190]
**Describe the bug**

I have a cluster configuration as follows : 

> root@ip-10-35-230-195:/usr/local/zeebe/bin# zbctl status --insecure
Cluster size: 3
Partitions count: 3
Replication factor: 3
Brokers:
  Broker 0 - 10.35.230.195:26501
    Partition 1 : Follower
    Partition 2 : Follower
    Partition 3 : Follower
  Broker 1 - 10.35.231.5:26501
    Partition 1 : Leader
    Partition 2 : Follower
    Partition 3 : Leader
  Broker 2 - 10.35.234.53:26501
    Partition 1 : Follower
    Partition 2 : Leader
    Partition 3 : Follower

I have one workflow instance created at partition Id 1.  Broker 1 is leader of that partition. 

I restart this node. Now the topology got changed as follows : 

> root@ip-10-35-230-195:/usr/local/zeebe/bin# zbctl status --insecure
Cluster size: 3
Partitions count: 3
Replication factor: 3
Brokers:
  Broker 0 - 10.35.230.195:26501
    Partition 1 : Leader
    Partition 2 : Follower
    Partition 3 : Follower
  Broker 1 - 10.35.231.230:26501
    Partition 1 : Follower
    Partition 2 : Follower
    Partition 3 : Follower
  Broker 2 - 10.35.234.53:26501
    Partition 1 : Follower
    Partition 2 : Leader
    Partition 3 : Leader

We can see broker 0 became leader of partition 1. 

Now if I try to submit fail command for that job, I'm getting following error:

> Caused by: io.zeebe.gateway.cmd.BrokerRejectionException: Command (FAIL) rejected (NOT_FOUND): Expected to fail activated job with key '2251799813735563', but it does not exist

**To Reproduce**

1. Create a workflow instance on the broker cluster
2. Check on which partition this workflow instance got created
3. Restart the node who is leader of that partition 
4. Once all the nodes are up again, try to run COMPLETE or FAIL command on that job 
 

**Expected behavior**

As we are replicating the partitions, I'm expecting command should not fail with "NOT_FOUND" error and the job should be available on other node. 

**Log/Stacktrace**
If possible add the full stacktrace or Zeebe log which contains the issue. 

<details><summary>Full Stacktrace</summary>
 <p>

```
io.zeebe.client.api.command.ClientStatusException: Command rejected with code 'FAIL': Expected to fail activated job with key '2251799813735563', but it does not exist\n\tat io.zeebe.client.impl.ZeebeClientFutureImpl.transformExecutionException(ZeebeClientFutureImpl.java:93)\n\tat io.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:50)\n\tat com.ihsmarkit.razor.handler.ZeebeWorkflowTaskHandler.handle(ZeebeWorkflowTaskHandler.java:63)\n\tat io.zeebe.client.impl.worker.JobRunnableFactory.executeJob(JobRunnableFactory.java:44)\n\tat io.zeebe.client.impl.worker.JobRunnableFactory.lambda$create$0(JobRunnableFactory.java:39)\n\tat java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)\n\tat java.util.concurrent.FutureTask.run(FutureTask.java:266)\n\tat java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)\n\tat java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\nCaused by: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: NOT_FOUND: Command rejected with code 'FAIL': Expected to fail activated job with key '2251799813735563', but it does not exist\n\tat java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)\n\tat java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)\n\tat io.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:48)\n\t... 10 more\nCaused by: io.grpc.StatusRuntimeException: NOT_FOUND: Command rejected with code 'FAIL': Expected to fail activated job with key '2251799813735563', but it does not exist\n\tat io.grpc.Status.asRuntimeException(Status.java:533)\n\tat io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:442)\n\tat io.grpc.PartialForwardingClientCallListener.onClose(PartialForwardingClientCallListener.java:39)\n\tat io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:23)\n\tat io.grpc.ForwardingClientCallListener$SimpleForwardingClientCallListener.onClose(ForwardingClientCallListener.java:40)\n\tat io.grpc.internal.CensusStatsModule$StatsClientInterceptor$1$1.onClose(CensusStatsModule.java:700)\n\tat io.grpc.PartialForwardingClientCallListener.onClose(PartialForwardingClientCallListener.java:39)\n\tat io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:23)\n\tat io.grpc.ForwardingClientCallListener$SimpleForwardingClientCallListener.onClose(ForwardingClientCallListener.java:40)\n\tat io.grpc.internal.CensusTracingModule$TracingClientInterceptor$1$1.onClose(CensusTracingModule.java:399)\n\tat io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:510)\n\tat io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:66)\n\tat io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:630)\n\tat io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$700(ClientCallImpl.java:518)\n\tat io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:692)\n\tat io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:681)\n\tat io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)\n\tat io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)\n\t... 3 more\n
```

</p>
</details>

**Environment:**

Version - 0.21.0-alpha-3
Running as docker image.
Embedded Gateway
Elastic Search exporter. 
Data folder mounted on external system for all the nodes.

**Comment by bhowmikdebarshi** [comment]
Sorry, I was trying to Fail the job. Updated the original post.

**Comment by deepthidevaki** [comment]
Did you activate the job before the node was restarted? I don't see it in "steps to reproduce".

**Comment by bhowmikdebarshi** [comment]
Not sure about activating the job. Do we need to do that manually ? 

One more thing I have noticed - 

Here is the simple workflow - 

![image](https://user-images.githubusercontent.com/22215160/66840455-b18ce300-ef85-11e9-8812-c65b09caefaf.png)


After creating the workflow instance, once it's waiting at message event, I restarted the node. 
Then I publish a message and noticed that the message got correlated successfully and moved to Process B. Here in Process B, it tried to Fail the job and got above exception.
